scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,369 @@
+                        .strip()\
+                        .replace(" ", "_")\
+                        .replace("__", "_")\
+                        for col in final_cols]
+
+        final_cols_list = computed_cols + final_cols_list
+        df.columns = final_cols_list
+
+        # Multiplying the data with scaling factor.
+        for col in final_cols_list:
+            if "count" in col.lower():
+                if self.scaling_factor != 1:
+                    df[col] = df[col].astype('float', errors="ignore").multiply(
+                        self.scaling_factor)
+                df[col] = df[col].astype('Int64', errors="ignore")
+        self.scaling_factor = 1
+
+        # Creating Location column with default value country/USA.
+        # as the dataset is all about USA country level only.
+        df.insert(1, "Location", "country/USA", True)
+        df.insert(0, 'date_range', _year_range(df['Date']), True)
+        return df
+
+    def _transform_data(self, df: pd.DataFrame) -> None:
+        """
+        This method calls the required functions to transform
+        the dataframe and saves the final cleaned data in
+        CSV file format.
+
+        Arguments:
+            file (str) : Dataset File Path
+
+        Returns:
+            df (DataFrame) : DataFrame.
+        """
+
+        # Finding the Dir Path
+        file_dir = os.path.dirname(self.cleaned_csv_file_path)
+        if not os.path.exists(file_dir):
+            os.mkdir(file_dir)
+        df = self._transform_df(df)
+        keep_value = "first"
+        if self.df is None:
+            self.df = df
+        else:
+            self.df = self.df.append(df, ignore_index=True)
+
+        self.df.sort_values(by=['Date', 'date_range'],
+                            ascending=False,
+                            inplace=True)
+        self.df.drop_duplicates("Date", keep=keep_value, inplace=True)
+        self.df.drop(['date_range'], axis=1, inplace=True)
+        self.df.to_csv(self.cleaned_csv_file_path, index=False)
+
+    def _generate_mcf(self, df_cols: list) -> None:
+        """
+        This method generates MCF file w.r.t
+        dataframe headers and defined MCF template
+
+        Arguments:
+            df_cols (list) : List of DataFrame Columns
+
+        Returns:
+            None
+        """
+        mcf_template = ("Node: dcid:{}\n"
+                        "typeOf: dcs:StatisticalVariable\n"
+                        "populationType: dcs:Person{}{}{}\n"
+                        "statType: dcs:measuredValue\n"
+                        "measuredProperty: dcs:count\n")
+        mcf = ""
+        for col in df_cols:
+            residence = ""
+            status = ""
+            armedf = ""
+            if col.lower() in ["date", "location"]:
+                continue
+            if re.findall('Resident', col):
+                residence = "\nresidentStatus"
+                status = ": dcs:USResident"
+            if re.findall('ArmedForces', col):

```

Comment:
Should these all be `if` or some `else if`?

Because you're overwriting residence/status